### PR TITLE
Create separate Address class; move data out of Company class

### PR DESCRIPTION
**Pull request by weedySeaDragon**
PT Story: Create an "Address" class; refactor out of company
https://www.pivotaltracker.com/story/show/140192663

This is the first

Note that I do _not_ remove the address-related columns out of the `companies` table in the db because we need to make sure that the current data is migrated out into the new `addresses` table first.  (I created the `shf:move_company_addresses_to_addresses` task for this.)

Changes proposed in this pull request:
1.  create the Address class (and associated addresses db table)
2. create the migration and move data out of the `companies` table and into the `addresses` table
3. get rid of address-related columns in the `companies` table
2. update `db:seed`
2. update specs and features
3. update CompanyController to load addresses for each company as appropriate
4. update views

Ready for review:
@patmbolger @thesuss

### CODE REVIEW

File: app/controllers/companies_controller.rb
Comment (patmbolger) [inline app/controllers/companies_controller.rb]: I think this statement could replace the above two statements:
```
@company.assign_attributes(company_params)
```
It is smarter about associated models when the parent model has `accepts_nested_attributes_for` (which your Company model _does_ have).

File: app/models/company.rb
Comment (patmbolger) [inline app/models/company.rb]: Is `foreign_key: :addressable_id` needed?  Should default to that, I think.

File: app/models/company.rb
Comment (patmbolger) [inline app/models/company.rb]: Since `Address` is polymorphic, and ID's can be the same across models (e.g. Company and User), shouldn't this be limited to addresses for companies?  Perhaps something like this:
```
have_no_regions = Address.lacking_region.where(addressable_type: 'Company').distinct.pluck(:addressable_id)
```

File: config/locales/en.yml
Comment (thesuss) [inline config/locales/en.yml:97]: Where is this used?

File: config/locales/en.yml
Comment (thesuss) [inline config/locales/en.yml:98]: Where is this used?

File: config/locales/en.yml
Comment (thesuss) [reply config/locales/en.yml:97]: Aha this one is "Region of operations"

File: config/locales/en.yml
Comment (thesuss) [reply config/locales/en.yml:98]: And this one is "Region"

**Review by thesuss — APPROVED** [review]
Nothing strange as far as I can see, but more people should have a look :)

File: lib/tasks/shf.rake
Comment (patmbolger) [inline lib/tasks/shf.rake]: This is a bit obscure(?).  Consider:
```
company.addresses << co_address
```

File: spec/factories/companies.rb
Comment (patmbolger) [inline spec/factories/companies.rb]: `num_addresses`

File: app/views/companies/_companies_list.html.haml
Comment (patmbolger) [inline app/views/companies/_companies_list.html.haml:37]: These statements will fail in production unless the rake task (included in this PR) is run first.  I assume this is the plan(?)

**Review by patmbolger — COMMENTED** [review]
Please see comments

**Review by thesuss — CHANGES_REQUESTED** [review]
Do not merge until rake task is run in production...

**Review by thesuss — APPROVED** [review]

File: app/models/company.rb
Comment (weedySeaDragon) [reply app/models/company.rb]: Good point.

File: app/controllers/companies_controller.rb
Comment (weedySeaDragon) [reply app/controllers/companies_controller.rb]: I couldn't make that work.  I wonder if it has something to do with the fact that we're dealing with an Array of Addresses for a Company?

File: lib/tasks/shf.rake
Comment (weedySeaDragon) [reply lib/tasks/shf.rake]: no longer applicable since that code is now deleted (not included in this PR)

**Review by patmbolger — APPROVED** [review]

**Pull request merged**
